src/main/java/org/mitre/synthea/modules/EncounterModule.java

```diff
@@ -44,19 +57,60 @@ public boolean process(Person person, long time) {
       Encounter encounter = person.record.encounterStart(time, EncounterType.WELLNESS.toString());
       encounter.name = "Encounter Module Scheduled Wellness";
       encounter.codes.add(ENCOUNTER_CHECKUP);
+      encounter.provider = person.getAmbulatoryProvider(time);
       encounter.codes.add(getWellnessVisitCode(person, time));
       person.attributes.put(ACTIVE_WELLNESS_ENCOUNTER, true);
+      encounter.reason = ENCOUNTER_CHECKUP;
       startedEncounter = true;
-    } else if (person.symptomTotal() > SYMPTOM_THRESHOLD) {
-      // add a symptom driven encounter if symptoms are severe
-      person.resetSymptoms();
-      Encounter encounter = person.record.encounterStart(time, EncounterType.WELLNESS.toString());
-      encounter.name = "Encounter Module Symptom Driven";
-      encounter.codes.add(ENCOUNTER_CHECKUP);
-      encounter.codes.add(getWellnessVisitCode(person, time));
-      person.attributes.put(ACTIVE_WELLNESS_ENCOUNTER, true);
-      startedEncounter = true;
-    }
+
+    } else if (person.symptomTotal() > EMERGENCY_SYMPTOM_THRESHOLD) {
+        if (person.attributes.get(LAST_VISIT_SYMPTOM_TOTAL) == null){
+          person.attributes.put(LAST_VISIT_SYMPTOM_TOTAL, 0);
+        }
+        if (person.symptomTotal() != (int)person.attributes.get(LAST_VISIT_SYMPTOM_TOTAL)) {
+          person.attributes.put(LAST_VISIT_SYMPTOM_TOTAL, person.symptomTotal());
+          person.addressLargestSymptom();
+          Encounter encounter = person.record.encounterStart(time, EncounterType.EMERGENCY.toString());
+          encounter.name = "Encounter Module Symptom Driven";
+          encounter.provider = person.getEmergencyProvider(time);
+          encounter.reason = ENCOUNTER_EMERGENCY;
```

Comment:
The reason for the encounter isn't that it is an emergency encounter.

If you can't turn the symptom into a code (hint: doesn't look like you can), then you should delete this line.